src/controllers/activitiesController.js

```diff
@@ -32,33 +32,32 @@ const saveActivity = async (req, res) => {
 // @access  public
 const updateActivity = async (req, res) => {
   const { _id } = req.params;
-  const { tally } = req.body;
-  if (!tally || isNaN(tally)) {
+  const { name, description, category } = req.body;
+  if (!name) {
     res.status(400);
-    throw new Error("An integer vote tally value must be provided!");
-  } else if (tally !== 1 && tally !== -1) {
+    throw new Error("New activity name must be provided!");
+  } else if (!category) {
     res.status(400);
-    throw new Error(
-      "Provide 1 to increase, and -1 to decrease the vote tally!"
-    );
+    throw new Error("New activity category must be provided!");
   }
 
-  const activity = await Activity.findOne({ _id });
-  if (activity) {
-    let vote = activity.votes;
-    if (vote > 0 || tally === 1) {
-      activity.votes = vote + tally;
-      await activity.save();
-    } else {
-      res.status(400);
-      throw new Error("Vote tally cannot be lowered than 0!");
-    }
-  } else {
+  const newActivityInfo = {
+    name,
+    description,
+    category,
+  };
+
+  const updatedActivity = await Activity.findOneAndUpdate(
+    { _id },
+    newActivityInfo,
+    { new: true, runValidators: true }
+  );
```

Comment:
Should've asked this on the `POST activities` route as well, but do we need to sanitize inputs?

Reply:
For the `POST` request route, during my testing the input validations set in the schema suffice. But for the `PUT` route, since `findOneAndUpdate` is used `runValidators: true` must be used in order to sanitize the inputs. Does that make sense?

Reply:
I can't think of any for these cases, so we can definitely skip it for now. Just a question for consideration (and maybe an easy fix with just a `sanitize` method of some sort.) But there may be problematic inputs which still pass validation.